pointcloud/basic_octree.go

```diff
@@ -35,7 +35,10 @@ type BasicOctree struct {
 type basicOctreeNode struct {
 	nodeType NodeType
 	children []*BasicOctree
+	parent   *BasicOctree
 	point    PointAndData
+	maxProb  float64
+	depth    int
```

Comment:
[q] Maybe this was discussed, but why add depth?  Is it storing the depth of a node in an octree or the depth of children that node has?  

Octrees have a cool recursive structure that we leverage such that each node is its own octree and this has the side effect of imparting information that may not be true if it is ever treated as a standalone octree down the road, so we should make sure that this is worth adding if we're going to break that abstraction

Reply:
I agree that if there is no current need for `parent` and `depth` we shouldn't include them. I can't currently think of a reason you'd need to 'back-out' of an octree one or two levels or know how deep you are in an octree, but I leave that up to the specifics in the ticket/project requirements.

Note: depth can always detemrined by `root_side*2^-depth = curr_side`

Reply:
will remove `depth` as a field and a add comment on how it can be calculated as
```
depth = root_side^2 - curr_side
```

Reply:
will also remove `parent`.